### PR TITLE
feat: shuffle function

### PR DESCRIPTION
**Pull request by vigan-abd**
### Description:
This PR adds function for shuffling arrays

### Breaking changes:
- [ ]

### New features:
- array shuffle

### Fixes:
- [ ]

### PR status:
- [x] Version bumped
- [x] Change-log updated
- [x] Tests added or updated
- [x] PR keeps 100% test coverage
- [x] Type definition updated
- [x] Readme updated
- [x] Linter is applied
- [x] Const arrow functions are used instead of pure function definitions where applicable
- [x] Functions are listed in alphabetic order in index.js, index.d.ts and readme

### CODE REVIEW

File: src/shuffle.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Performs pseudo random shuffles on clone of the array
+ *
+ * @param {Array<any>} array
+ * @returns {Array<any>}
+ */
+const shuffle = (array) => {
+  const clone = [...array]
+  const length = clone.length
+  for (let i = 0; i < length; i++) {
+    const mov = Math.floor(Math.random() * length)
```
Comment (jarco-develop) [inline src/shuffle.js]: I think this shuffle function may not provide an even distribution of permutations, as you're always picking from the entire range of the array for every position.

May I suggest to this approach: 

´´´js
...
for (let i = length - 1; i > 0; i--) {
  const index = Math.floor(Math.random() * (i + 1)); // random index from 0 to i
  [clone[i], clone[index]] = [clone[index], clone[i]]; // swap elements
}
...
´´´

Comment (dictor93) [reply src/shuffle.js]: Richard Durstenfeld algorithm?

**Comment by dictor93** [comment]
lgtm

**Review by dictor93 — APPROVED** [review]

File: src/shuffle.js (hunk 1 above)
Comment (jarco-develop) [reply src/shuffle.js]: exactly 🚀

**Review by jarco-develop — APPROVED** [review]
LGTM 🐐

**Pull request merged**